src/datasets/arrow_dataset.py

```diff
@@ -1645,7 +1645,8 @@ def init_buffer_and_writer():
         if update_data and tmp_file is not None:
             tmp_file.close()
             shutil.move(tmp_file.name, cache_file_name)
-            os.chmod(cache_file_name, 0o644)
+            umask = os.umask(0)
```

Comment:
While getting the umask value, you are also setting the os.umask to zero here.

First, it appears that setting the value to zero might a [security issue](https://stackoverflow.com/a/44130549) so please use `0o666` instead.

Then, you can set it back to the original value.

```suggestion
            umask = os.umask(0o666)
            os.umask(umask)
```

Reply:
TIL ^^